src/wasm.rs

```diff
@@ -0,0 +1,301 @@
+//! # BLS12-377 WASM Bindings for Blind Threshold Signatures.
+use wasm_bindgen::prelude::*;
+
+use rand_chacha::ChaChaRng;
+use rand_core::{RngCore, SeedableRng};
+
+use crate::{
+    curve::zexe::PairingCurve as Bls12_377,
+    group::{Element, Encodable, Point},
+    sig::{
+        blind::{BG1Scheme, Token},
+        tblind::G1Scheme,
+        Blinder, Scheme, SignatureScheme, ThresholdScheme,
+    },
+};
+
+type BlindThresholdSigs = G1Scheme<Bls12_377>;
+type BlindSigs = BG1Scheme<Bls12_377>;
+type PublicKey = <BlindThresholdSigs as Scheme>::Public;
+type PrivateKey = <BlindThresholdSigs as Scheme>::Private;
+type Signature = <BlindThresholdSigs as Scheme>::Signature;
+type Result<T> = std::result::Result<T, JsValue>;
+
+/// Signatures for BLS12-377 are 197 bytes long
+const SIG_SIZE: usize = 197;
+
+///////////////////////////////////////////////////////////////////////////
+// User -> Library
+///////////////////////////////////////////////////////////////////////////
+
+#[wasm_bindgen]
+/// Given a message and a seed, it will blind it and return the blinded message
+///
+/// # Safety
+/// NOTE: If the same seed is used twice, the blinded result WILL be the same
+pub fn blind(msg: Vec<u8>, seed: &[u8]) -> BlindedMessage {
+    let mut rng = get_rng(&seed);
+    let (scalar, blinded_message) = BlindThresholdSigs::blind(&msg, &mut rng);
+
+    BlindedMessage {
+        message: blinded_message,
+        scalar,
+    }
+}
+
+/// Given a blinded signature and the scalar used to blind the original message, it will return the
+/// unblinded signature
+///
+/// If unmarshalling the scalar or unblinding the signature errored, it will return an empty
+/// vector
+#[wasm_bindgen]
+pub fn unblind_signature(
+    blinded_msg: Vec<u8>,
+    scalar: *const Token<PrivateKey>,
+) -> Result<Vec<u8>> {
+    let scalar = unsafe { &*scalar };
+
+    BlindThresholdSigs::unblind(&scalar, &blinded_msg)
+        .map_err(|_| JsValue::from_str("could not unblind signature"))
+}
+
+#[wasm_bindgen]
+/// Verifies the signature after it has been unblinded. Users will call this on the
+/// threshold signature against the full public key
+pub fn verify_sign(public_key: *const PublicKey, msg: Vec<u8>, signature: Vec<u8>) -> Result<bool> {
```

Comment:
How about we return an error in case of bad verification? IMO APIs that return bool are easy to ignore.